### PR TITLE
test: add cypress smoke test to UI tests

### PR DESCRIPTION
**Pull request by romainverduci**

### CODE REVIEW

**Comment by romainverduci** [comment]
New tests are running there: https://github.com/jenkins-x/jx/pull/6060
Known issue: junit report aggregation is broken and will be fixed once the builder includes https://github.com/jenkins-x/jx/pull/6087

**Comment by romainverduci** [comment]
From the build logs:
```
> ui-smoke-tests@1.0.0 test /tmp/romainverduci/bdd-jx/test/suite/apps/ui-smoke
> cypress run

It looks like this is your first time using Cypress: 3.6.0

[13:43:13]  Verifying Cypress can run /root/.cache/Cypress/3.6.0/Cypress [started]
[13:43:14]  Verified Cypress!       /root/.cache/Cypress/3.6.0/Cypress [title changed]
[13:43:14]  Verified Cypress!       /root/.cache/Cypress/3.6.0/Cypress [completed]

Opening Cypress...

tput: No value for $TERM and no -T specified
================================================================================

  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    3.6.0                                                                              │
  │ Browser:    Electron 73 (headless)                                                             │
  │ Specs:      1 found (smoke.spec.js)                                                            │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────
                                                                                                    
  Running:  smoke.spec.js                                                                   (1 of 1)

  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        6                                                                                │
  │ Passing:      6                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     5 seconds                                                                        │
  │ Spec Ran:     smoke.spec.js                                                                    │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


================================================================================

tput: No value for $TERM and no -T specified
  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped  
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  smoke.spec.js                            00:05        6        6        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:05        6        6        -        -        -  

Moving test report to /workspace/source/build/reports
```

File: test/suite/apps/ui.go [hunk 1]
```diff
@@ -148,6 +149,34 @@ func (t *AppTestOptions) UITest() bool {
 			Expect(err).ShouldNot(HaveOccurred())
 		})
 
+		It("Runs smoke tests", func() {
```
Comment (hferentschik) [inline test/suite/apps/ui.go]: Nice :-)
How long do these tests take to run?

File: go.mod [hunk 2]
```diff
@@ -23,6 +23,7 @@ require (
 	github.com/chromedp/chromedp v0.1.1
 	github.com/codeship/codeship-go v0.0.0-20180717142545-7793ca823354
 	github.com/denormal/go-gitignore v0.0.0-20180713143441-75ce8f3e513c
+	github.com/docker/docker v0.0.0-20171206114025-5e5fadb3c020
```
Comment (hferentschik) [inline go.mod]: Docker?

File: test/suite/apps/ui-smoke/.npmrc [hunk 3]
```diff
@@ -0,0 +1 @@
+scripts-prepend-node-path=true
```
Comment (hferentschik) [inline test/suite/apps/ui-smoke/.npmrc]: new lines missing

**Comment by romainverduci** [comment]
> Looking good.
> 
> I am wondering about the amount of time it takes though to run the tests. I would expect that the installation and all takes much longer than these smoke tests, right? However, given how long the BDD tests already take we might have to consider soon where and how BDD tests run.

According to the report it took 66 seconds to run (including installing cypress from npm and set it up which takes some time).
The tests themselves took something around 6 seconds so if we add more tests it won't be so bad.

File: go.mod (hunk 2 above)
Comment (romainverduci) [reply go.mod]: Pushed by mistake, don't know how it's been upgraded :/

**Comment by hferentschik** [comment]
/test pr-build

not sure what the issue is with this status check. Might be due to the current cluster issues?

**Pull request merged**